### ISSUE TITLE
more informative initial screen

### ISSUE THREAD

**Issue by mfrasca** [Bauble/bauble.classic#224]
the initial `DefaultView` is something which scared me off terribly, the first time I ever saw Bauble. I tend to call it the "what do I do now" screen. I am considering, also loosely connected to #132, to add some stuff in it, something like this:
![screenshot from 2015-12-18 16 52 50](https://cloud.githubusercontent.com/assets/122869/11908167/128135ac-a5a8-11e5-8741-4cf021e2e410.png)

the texts would link to searches, like `xxx plants` associated to `plants like %`.

I'm just playing with the idea as of now...
maybe a second and third line with information about those plants/accessions which still await complete identification:
`%(p)s plants in %(a)s accessions, identified at rank genus, in %(g)s genera.`,
`%(p)s plants in %(a)s accessions, identified at rank family, in %(f)s families.`, 

maybe also the ones identified at rank species could follow the same format.

**Comment by mfrasca** [comment]
and yes, it should go in the glade file, not in the code! :crocodile:

**Comment by RoDuth** [comment]
Would be great to have a few saved searches here also.  i.e. You could save your favourite 5 or 10 search strings with names that make sense to you and others users, it then becomes available on this screen as a hyperlink or something.  Maybe you could have a "clear search results" button next to the search button that if you press returns you here to use these functions again.  Rather than have to close down and restart.  Just thinking out loud...

**Comment by mfrasca** [comment]
have been fighting against gtk not liking `functools.partial`, why not, no idea. anyhow: this is how it looks like now. and yes it now becomes necessary offering the option to get back to the splash screen.
![screenshot from 2015-12-19 22 05 02](https://cloud.githubusercontent.com/assets/122869/11916266/e347ce90-a69c-11e5-80c7-169be8a29cce.png)

the numbers are clickable and the mouse was on top of **genera in use**

I have to work at the formatting, for sure. going to commit anyhow.

**Comment by mfrasca** [comment]
except the fact we miss unit tests (which I cannot add until I solve #132) and not being able to get back to this screen, also linked to #132, I can consider this issue closed.
feel free to comment and reopen.
![screenshot from 2015-12-20 08 48 37](https://cloud.githubusercontent.com/assets/122869/11918231/85a1c2dc-a6f6-11e5-8d92-b707d02be395.png)

**Comment by mfrasca** [comment]
notice the :house: icon. I am afraid of cluttering the home screen though.
must find a balance between "what do I do now" and "so many trees I can't see the wood"
![screenshot from 2015-12-20 11 32 56](https://cloud.githubusercontent.com/assets/122869/11918783/747d4ef6-a70d-11e5-82ec-8e8c84deac2f.png)
I think the next steps are on #36 and I suggest we continue the discussion there.

**Comment by RoDuth** [comment]
Wow @mfrasca that looks great!  I think you got the balance just right.  Looking forward to seeing it in action.

**Comment by mfrasca** [comment]
what is missing...
updating the counts if you insert something while this screen is is active. (unreported)
all of the 'stored queries' handling. #36
cleaning the status bar and the search text when you click the home button. (unreported)

**Comment by tmyersdn** [comment]
@mfrasca @RoDuth, this new content on the initial page looks really great

(1) What about using a "refresh" symbol instead of "home", or have I misunderstood what home does?

(2) How about saying "alive" instead of "used" on the table?

(3) Would it be possible to add lines for "number of propagations" and "number of voucher specimens"? (and ideally "number of seed in storage" for seed list, "seed orders" made from seed list)

(4) How about a line for number of "tags" that are active?

(5) Do these numbers need to be static, or can they be clickable to generate a report (obviously a silly question, as most would be redundant and too long, but it could be useful for tags.

(6) Is there some way for issues needing attention to be flagged to the user at initial screen, perhaps tags could be used this way?

Okay, I realize I may be way off track here, just trying to get my head around what you two are interested in with the initial screen. I also am keen that the initial screen has prompts to help and interest people into using Bauble.

**Comment by RoDuth** [comment]
@tmyersdn My thoughts on your questions..
1) The home button returns you to the initial screen if you are in a search, which is where you normally are.  It used to be that you would see this initial screen only once per season because as soon you searched for anything it was replaced by the search results.  I guess you could call this initial screen the "home" screen.
2) Semantics? Got to say I did question it at first but I don't know if "52 alive families" made any more sense to me in the context.  I'm happy to see it as I have 514 families in my database but I'm only "using" 52 of them.  And "alive locations"? "Alive plants" does work but the others not so much. I'm actually interested in the idea of having an "Unused" or "Empty" column, I know that its a little "double handling" but because they are hyper-links to the relevant search I can see it being very useful.
3) I like the idea, how about "Vouchered Accessions" rather than "Vouchers" (some of ours have 2 or even 3 against them!!!). Could be based of `accession where vouchers.id != 0` and the inverse "Unvouchered Accessions" `accession where vouchers = Empty`.  This I would definitely use. Then "Plants Propagated From" with `plant where propagations.id != 0`.  Would mess with the formatting a bit as they wouldn't fit the grid but they could just sit as three extra elements below it maybe?  Not sure I can see a neat way to do the seed one though.  But we don't currently accession seeds here.
4) I kind of like tags where they are in the menus and I'm not sure how you would implement this considering the amount of different ways to use tags.
5) Give it a go Tom, they are all hyper-links to the relevant searches!   I love this side of Bauble, point around and anything that turns blue when you hover over it (as all the numbers do in this view) will take you to the relevant data. (try it in other views also).  This is one of the main things I am "interested in with the initial screen" as you asked.  With this simple list you immediately have 10 of the most common searches to get into the data, just one click away.  This is why I was thinking that an "unused" (or "Empty") column would be handy also, it would add another 4 searches (species, accessions, plants, locations - families and genera being greyed out) that would be very handy to me at least.
6) I just use tags, as you say, for this and use them from the menu.  As I said before, not sure how you would make this work...  You could, of course, always create a tag called "Needing Attention" and (in the future) set one of the "stored queries" buttons up to display them?

Your not off track, some good points Tom.  

I'm waiting to see how Mario implements the "Stored Queries" buttons...  That's something that could be very handy at making it immediately user friendly to those who struggle with writing search strings and just make it faster to get results out of in general.

While I get what you mean about the prompts to interest people I would hate for Bauble to become as patronising, plain annoying and cluttered as some applications I have seen (remember "Clippy" the paper-clip office assistant!  Aaargh!).  I think, like any software, you need to spend some time exploring it and I think that good documentation (which Mario has made major improvements on) is key here.

**Comment by mfrasca** [comment]
I chose 'used' after a first attempt with 'used / alive', but the broader header cluttered the table. I said to myself: it's just an indication, then users get used to it and do not even take notice of the text there.
@RoDuth , one of your comments became #244 . thanks!
an other part of your above comment, I will put it as part of #36 if you're not faster than me.
for the #36 Stored Queries I think I first want to implement #184 Functions in Search, in particular I think you will be quite happy with date calculations (on right hand side) and max/count aggregating functions on left hand side of binary test functions (=, !=, <, etc).

**Comment by mfrasca** [comment]
and I agree with @RoDuth : tags are such a strange flexible beast that users can take advantage of in so many different ways... you can group, under the same tag, things of different order like a location and a family and in theory just about anything, it's the user interface that puts some limits to it.
(if you have a "Needing Attention" tag, you type 'needing attention' -with quotes- as a search and you get it)

**Comment by tmyersdn** [comment]
thanks both of you, yes, I need to spend more time trying out the functions

1) thanks for explaining the home button, I do find the home button a little disconcerting, but can get over that. But how about the refresh symbol (single or double arrows in a circle) or the start/power symbol (open circle with vertical line)?
 2) Used vs other - okay, I get your logic and see the difficulty (I don't mind living families, but locations are not living!). How about "active", or Mario's earlier version of "In use".  I find "used" to sound vaguely negative.  An unused column sounds okay, but I don't see why it is needed, unless you want rapid access to the hyperlinked searches? what is your thinking here? (I can see that unused names or locations might be useful, but not unused families?)

 3) vouchers
- the problem for us is that we have a rather eclectic set of vouchers, some are from weeds (not accessed), and wild plants (also not accessed, usually outside the garden, but sometimes associated with collected seed).  So, I am curious to know if Bauble can handle vouchers that are not accessed? Or alternatively, if a rule is needed that all vouchered plant material gets an accession number, a provenance, and also a flag to say the source plant is not part of the living collection?
- we have the same problem for seed, and I guess for images.  Not so for propagations, where all incoming propagations get a new accession number (with the exception of some short term display crops, which staff refuse to access).

Even so, I still like your "accession where vouchers.id != 0"  and the inverse and "plants where propagations.id != 0".  Both good tools to show which plants have been overlooked for propagation or vouchering.  We also have a fair number of plants growing in our nursery that are ready to plant, and it would be good to be able to hyperlink to these from the home screen.

We collect seed of approximately 90 species per year, clean and store it, produce a seed list, and record and fill orders from other gardens.  Then each year we start all over again with new seed. It would be very useful to know how many orders are currently waiting to be processed and sent out.  It would also be useful to show where all the seed was sent to.

4) tags - okay -  how about having another column beside "total" and "used" for "number of tagged items" in the home page table?

I will confess, I am struggling with tags, I can create a tag in the tags select menu tool, and I can see the tags window below the search results window, but anything I type in here just disappears. Part of my problem is that our laptop is not networked, so I can't access the online help. Is there a link you can refer me too? I have used tags more successfully when I first tried using Bauble a few years ago, so it is something I have forgotten how to do.
But I agree that tags are extremely useful.

5) hah - I always like it when I request something that already there! Okay, for lazy readers like me, how about embedding the numbers into button symbols (same as for stored queries).

6) thanks, I am very happy to use tags in this way :)

Okay you two, @mfrasca & @RoDuth, many thanks for your hard work this year. I hope you have a great new year - will catch you in a few days :)

**Comment by RoDuth** [comment]
Sorry @tmyersdn missed this post, no doubt because the issue is closed :confounded:  ...  2 of your issues are not ones I have tried to tackle as of yet but can see a day when I may need to... seed banking and herbarium.  

We have only just recently started doing our own propagation and on a very minor scale at this point.  We don't have a seed bank of any form and our herbarium is completely volunteer run by people that are... not of the computer generation :smile: ... and not inclined to want anything to do with databases etc.  They do an amazing job and keep really good records that, at this point, are completely separate to ours and are hand-written! and I cant see a point any time soon that this will change.  I just plain don't have the time.  We only ever need to store the voucher number against the accession, anything else they voucher may be in their register but if it is not from the living collection its not of any concern to us.   Don't forget I have very limited time for the records side of my job so I have to prioritise (everything you see of me here is in my own time.)  A classic example of the differing needs of gardens.

As for the others issues.. 

2) Sorry I thought I had said families and genera would be greyed out (unusable in the search respect as is the case for them in the total column, maybe @mfrasca edited this out to make #224?) I was thinking about the fact that the searches would get you to unused accessions, species, plants and locations... all very relevant searches especially if you are looking for things that may be wrong.  When I saw that I had 4 unused accession and 1 unused plants it made me wonder how that could be.  It was just a thought.  I still like it in principal.

4) tags... just select anything on the left hand side of a search result (a location, accession, species, plant, family, anything really just make sure the line or lines are orange) then go to the "Tags" menu and select "Tag Selection", if you don't have any tags yet or need a new one click the "+New Tag" button on the bottom left of the popup window and create a new tag with description etc. then press "OK" and return to the tags window.  Now select (tick box to left) whichever tags you would like to apply to the selection you have chosen.  Simple!  And really useful how you can combine unrelated data in this way.  To see everything that has been tagged with a certain tag just select the tag name from the "Tags" menu and it will run the `tag="foo"` search for that tag. (or just type it yourself of course)  Because of the way tags can apply across all levels and the fact that any one tagged item can have multiple tags I don't see any use for a "tagged" column as you suggest.

5) I think that once you are used to this being the way things are throughout Bauble you don't even question it... Like the first time I use a new version of MS Windows or OSX etc.... I have no idea how to even navigate it at first now its all second nature...  The sort of thing that good documentation covers.  Also if you start enclosing these hyper-links in buttons then where do you stop?  This functionality is all through Bauble.  e.g. Select an accession from a search then in the right hand info pain you can click its species name to go to the species data (and, using the triangles to expand, anything attached)

**Comment by tmyersdn** [comment]
@RoDuth, thanks, yes another comment in a closed issue :confounded:, but I cant be bothered reopening it, and I know you will see this sooner or later.

Completely agree with the need to keep what you like in Bauble/Ghini.  Ah, Tags, I can see this is something that I will come back to talk to you and Mario about later.

I have partly been asking idiot questions to help focus on "new user" experience.  "the way things are" approach I believe is alienating to managers and people who decide to install them. Ie people in charge of records officers.

**Comment by mfrasca** [comment]
I have modified a bit "the way things are"... I've experimented in the home screen with a border around a sensitive area, and the background of the sensitive area becomes slightly lighter when you hover on it.
experimenting, meaning that you will see three different behaviours then you tell me what you like most.

border, lighter background and foreground turning blue on hovering. (home screen)
no border, lighter background and foreground turning blue on hovering. (most of the links in the info-boxes)(these will replace the selection)
no border, foreground turning blue on hovering (genus and family link in species and genus info boxes)(these will add to the selection)

I think I like it this way. but you may convince me of the contrary.
![screenshot from 2016-01-20 17 36 06](https://cloud.githubusercontent.com/assets/122869/12465500/53ad6538-bf9c-11e5-855e-4335ef87920b.png)

**Comment by mfrasca** [comment]
(my proverbially bad memory makes me a persistently new user of everything)

**Comment by tmyersdn** [comment]
yes, me too Mario - thanks for posting the new screen shot, I like the new Ghini image.  You know he is holding an herbarium specimen! (so I will make my case for removing some of the empty report rows and adding "vouchers" to the table!).... :)

**Comment by mfrasca** [comment]
if I know what he is holding! :laughing: a friend of mine made the drawing on my kind request, and I digitalized it. 
https://www.facebook.com/bauble.thesoftware/photos/pb.335471699996076.-2207520000.1453330722./406027922940453/?type=3&theater

**Comment by RoDuth** [comment]
I like it, looks good just the way it is now.  And I do really like the unused column.  I seem to be using it quite a bit lately.  For instance, when we are putting together a species list for planting we have started using the database to do this adding the accession with intended location (although 2 intended locations isn't always enough so I may add notes etc.  @mfrasca is there a way that this could be just a list of intended locations and still be able to search it?) and no received date.  Then when we receive it the date is added and when it is planted a plant is attached.  Clicking the unused accessions gets me quickly to all those plants we are intending to plant.    Also, clicking unused plants gets you to the dead plants.

@tmyersdn Out of interest one of the first queries I added to the saved queries was `accession where vouchers.herbarium = "MRBG"` to find all those that accessions that we have vouchered!  Not quite the same as what you are suggesting.  I still think that if I was to start recording our herbarium in Bauble (as in those that don't relate to a plant in the ground) I would run a second database (and connection) for the specific purpose of just recording the herbarium.

@tmyersdn I do appreciate the new user perspectives and am sure @mfrasca does too.  I think we have definitely gained from them.  I like how @mfrasca's slight changes with the light background and thin outlining draws attention to the hyper-link functionality for a new user which then makes them aware of it else where within the other views.  (i.e. the species hyper-link from an accession view turns blue when you hover over it and you recognise that function from this front screen and therefore recognise it as a hyper-link to click, but it doesn't need to have the whole light background and outlining any more)  That's a great subtle change that satisfies my desire for the simple and elegant look and feel and yours for the new user experience in my opinion.

My whole thing is maintaining the uniqueness, the points of difference, of Bauble/Ghini and yes sometimes that means that there will be things new users will need to get used to.  Kind of like the first time anyone put push button, electric windows in a car, I bet people hated it (I did for one!) and were looking for the winding handle but now days everyone is used to and enjoys them and the old winding handle seems archaic...  To me, because I am used to it, having to show someone that you can click on something by making it look like a button... seems archaic.

**Comment by tmyersdn** [comment]
Thanks @RoDuth, @mfrasca
I realize behind all the front end changes there is a lot of work, and also that it may detract from other issues that also need attention.  I would prefer rather than just getting a blanket "no" on these suggestions, that they get parked for now for later discussion. 

My interest in this front screen work is that this is the screen to attract new users (gardeners and garden managers). This summary table (in my opinion) is quite unique to Ghini, and should be attractive particularly to people who would normally stay away from databases. It is a form of database "dashboard".  Ross, this is also behind my interest in tags, I understand you like them and do not want them messed with. What I am getting at here is that I think the "tag" function will take on a whole new dimension for a non-database using manager, who will see a way of flagging garden issues. Based on this, I suggested a column of tags counts for plants, species etc. Yes, I know the same tag can be applied to multiple data types, but this would be counted against each row.

Mario, could the used and unused columns be combined in one column (Eg "Species 247 / 66") and still keep hyperlinks for each number?  Not very elegant perhaps.. (fails the Ross test).

Finally, yes, I know that Mario is already swamped with work. Mario, apologies, you have closed this issue already, and I keep adding to it..

@RoDuth :bow: thanks for trying that voucher report. Yes, you are right, this could be a separate database and connection.  But obviously it would have similar name issues to Ghini and would still need to include garden plants.

**Comment by mfrasca** [comment]
> I would prefer rather than just getting a blanket "no" on these suggestions, that they get parked for now for later discussion. 

the best way (the way that would reduce the amount of work for me) is if we could isolate issues very fine grained. so I do not need question myself, after solving an issue (the abstract concept) and before closing the issue (the github object): "are there more issues in this issue?"

I'm going to open something related to tags, it has zero impact on the database so it does not get into the ghini-1.1.1 milestone.

**Comment by tmyersdn** [comment]
:imp: will create fine grained issues

**Comment by mfrasca** [comment]
:+1:
